controllers/configurationpolicy_utils.go

```diff
@@ -307,14 +307,19 @@ func formatMetadata(metadata map[string]interface{}) (formatted map[string]inter
 		md["labels"] = labels
 	}
 	if annos, ok := metadata["annotations"]; ok {
-		noAutogenerated := map[string]interface{}{}
-		for key := range annos.(map[string]interface{}) {
-			if !isAutogenerated(key) {
-				noAutogenerated[key] = annos.(map[string]interface{})[key]
+		// When a non-map is provided, skip iterating and set the value directly
+		if iterableAnnos, ok := annos.(map[string]interface{}); ok {
+			noAutogenerated := map[string]interface{}{}
+			for key := range iterableAnnos {
+				if !isAutogenerated(key) {
+					noAutogenerated[key] = annos.(map[string]interface{})[key]
+				}
 			}
-		}
-		if len(noAutogenerated) > 0 {
-			md["annotations"] = noAutogenerated
+			if len(noAutogenerated) > 0 {
+				md["annotations"] = noAutogenerated
+			}
+		} else {
+			md["annotations"] = annos
```

Comment:
What's the thought process on including an invalid annotation?

Reply:
So that an error is returned in the status. I don't think we validate elsewhere, and this matches the behavior of the `Labels` key above.